### PR TITLE
Group detail page - Add/Remove policies

### PR DESCRIPTION
**Pull request by lgalis**
1. Separate the policy step in 2 substeps, one for the name/description, the second for adding roles
Screenshot from 2019-09-12 15-46-12

Screenshot from 2019-09-12 15-46-45

2. Added async filtering to the role selection.
Screenshot from 2019-09-12 16-58-44

3. Group detail page - policies add/edit/remove

Screenshot from 2019-09-12 12-52-32

4. Add policy wizard
Screenshot from 2019-09-15 22-40-54

Next steps: Edit group polices. Replace the Select component with a compact list - to follow in separate PRs

### CODE REVIEW

File: src/smart-components/group/policy/add-policy/summary-content.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  GridItem,
+  Stack,
+  StackItem,
+  Text,
+  TextContent,
+  TextVariants,
+  Title
+} from '@patternfly/react-core';
+
+import {  } from '@patternfly/react-core';
```
Comment (Hyperkid123) [inline src/smart-components/group/policy/add-policy/summary-content.js]: Empty import

File: package.json [hunk 2]
```diff
@@ -14,12 +14,13 @@
     "@redhat-cloud-services/rbac-client": "^1.0.31",
     "awesome-debounce-promise": "^2.1.0",
     "axios": "^0.18.0",
+    "node-fetch": "^2.6.0",
```
Comment (Hyperkid123) [inline package.json]: This library does not seem to used anywhere.

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js [hunk 3]
```diff
@@ -1,4 +1,111 @@
...
 
-export default AddPolicyWizard;
+AddGroupPolicyWizard.propTypes = {
+  history: PropTypes.shape({
+    goBack: PropTypes.func.isRequired
+  }).isRequired,
+  addNotification: PropTypes.func.isRequired,
+  createPolicy: PropTypes.func.isRequired,
+  postMethod: PropTypes.func,
+  inputValue: PropTypes.string,
```
Comment (Hyperkid123) [inline src/smart-components/group/policy/add-policy/add-policy-wizard.js]: This is never used

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js [hunk 4]
```diff
@@ -1,4 +1,111 @@
...
+  match: PropTypes.object,
+  closeUrl: PropTypes.string
+};
+
+const mapStateToProps = ({ roleReducer: { roles, filterValue, isLoading }}) => ({
+  roles: roles.data,
+  pagination: roles.meta,
+  isLoading,
+  searchFilter: filterValue
```
Comment (Hyperkid123) [inline src/smart-components/group/policy/add-policy/add-policy-wizard.js]: this is never used

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js [hunk 5]
```diff
@@ -1,4 +1,111 @@
...
+});
+
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+  addNotification,
+  fetchGroup,
```
Comment (Hyperkid123) [inline src/smart-components/group/policy/add-policy/add-policy-wizard.js]: never used

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js [hunk 6]
```diff
@@ -1,4 +1,111 @@
...
+  createPolicy,
+  fetchRoles
```
Comment (Hyperkid123) [inline src/smart-components/group/policy/add-policy/add-policy-wizard.js]: never used

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js [hunk 7]
```diff
@@ -1,4 +1,111 @@
...
+      variant: 'warning',
+      title: 'Add policy',
+      description: 'Adding policy was cancelled by the user.'
+    });
+    push(closeUrl);
+  };
+
+  return (
+    <Wizard
+      isLarge
```
Comment (Hyperkid123) [inline src/smart-components/group/policy/add-policy/add-policy-wizard.js]: This is not a valid prop for a wizard

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js (hunk 6 above)
Comment (lgalis) [reply src/smart-components/group/policy/add-policy/add-policy-wizard.js]: it is used in fetchData

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js (hunk 7 above)
Comment (lgalis) [reply src/smart-components/group/policy/add-policy/add-policy-wizard.js]: removed

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js (hunk 5 above)
Comment (lgalis) [reply src/smart-components/group/policy/add-policy/add-policy-wizard.js]: removed

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js (hunk 4 above)
Comment (lgalis) [reply src/smart-components/group/policy/add-policy/add-policy-wizard.js]: removed

File: src/smart-components/group/policy/add-policy/add-policy-wizard.js (hunk 3 above)
Comment (lgalis) [reply src/smart-components/group/policy/add-policy/add-policy-wizard.js]: removed

File: package.json (hunk 2 above)
Comment (lgalis) [reply package.json]: There were errors on the travis test indicating this was missing, so I've added it - I will try again without it.

File: src/smart-components/group/policy/add-policy/summary-content.js (hunk 1 above)
Comment (lgalis) [reply src/smart-components/group/policy/add-policy/summary-content.js]: removed

File: package.json (hunk 2 above)
Comment (lgalis) [reply package.json]: Tried again without this library - while my local testing works, the travis build fails with these type of errors:
FAIL  src/test/presentional-components/shared/pf4-select-wrapper.test.js

  ● Test suite failed to run

    Cannot find module 'node-fetch' from 'server.js'

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:229:17)

      at Object.<anonymous> (node_modules/fetch-mock/src/server.js:1:104)

Comment (lgalis) [reply package.json]: I added the node-fetch back.

**Comment by lgalis** [comment]
@Hyperkid123 - feedback addressed

File: package.json (hunk 2 above)
Comment (Hyperkid123) [reply package.json]: Oh i see. Does this app still use fetch/fetch-mock somewhere? That is the source of this issue. I will do some audit once i am back from PTO.

**Pull request merged**